test/grouping.jl

```diff
@@ -1458,4 +1456,42 @@ end
     @test_throws KeyError gd[(a=:A, b=:X)]
 end
 
+@testset "Allow returning DataFrame() or NamedTuple() to drop group" begin
+    for i in typemin(UInt8):typemax(UInt8),
+        er in (DataFrame(), view(DataFrame(ones(2,2)), 2:1, 2:1),
+               view(DataFrame(ones(2,2)), 1:2, 2:1),
+               NamedTuple(), rand(0,0), rand(5,0))
```

Comment:
Maybe also test `DataFrame(a=Int[])`, `DataFrame(a=Any[])`, `NamedTuple(a=Int[])` and `NamedTuple(a=Any[])`?

Reply:
OK

Reply:
I understand you meant `x1` instead of `a` here?

Reply:
added also `rand(0, 1)` as it is a similar special case.